File: lowfare-flights-search-client/src/app/services/flight-offer/flight-offer.ts
import {Price} from "./price";
import {Airline} from "./airline";

export class FlightOffer {
  id: string;
  origin: string;
  destination: string;
  numberOfTransfers: number;
  departureDate: string;
  returnDate: string;
  numberOfPassengers: number;
  price: Price;
  airline: Airline;
  departure: Date;
  arrival: Date;
}

File: lowfare-flights-search-service/src/main/java/hr/msutic/lowfareflights/flightoffer/FlightOffer.java
package hr.msutic.lowfareflights.flightoffer;

import hr.msutic.lowfareflights.airline.Airline;
import lombok.Builder;
import lombok.Getter;
import lombok.ToString;

import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * @author Marko
 */
@Getter
@Builder
@ToString
public class FlightOffer implements Serializable {

  private String id;
  private String origin;
  private String destination;
  private LocalDate departureDate;
  private LocalDateTime departure;
  private LocalDate returnDate;
  private LocalDateTime arrival;
  private int numberOfPassengers;
  private int numberOfTransfers;
  private Price price;
  private Airline airline;

}
